cfme/tests/cli/test_appliance_console.py

```diff
@@ -269,81 +276,88 @@ def is_storage_extended():
+                   ipa_crud.ipadomain or CLEAR,
+                   ipa_crud.iparealm or CLEAR,
+                   ipa_crud.ipaprincipal or CLEAR,
+                   ipa_crud.bind_password,
+                   TimedCommand('y', 40), CLEAR)
     configured_appliance.appliance_console.run_commands(command_set)
+    configured_appliance.sssd.wait_for_running()
+    assert configured_appliance.ssh_client.run_command("cat /etc/ipa/default.conf |"
+                                                       "grep 'enable_ra = True'")
 
-    def is_sssd_running(configured_appliance):
-        assert configured_appliance.ssh_client.run_command("systemctl status sssd | grep running")
-    wait_for(is_sssd_running, func_args=[configured_appliance])
-    return_code, output = configured_appliance.ssh_client.run_command(
-        "cat /etc/ipa/default.conf | grep 'enable_ra = True'")
-    assert return_code == 0
+    # Unconfigure to cleanup
+    # When setup_ipa option selected, will prompt to unconfigure, then to proceed with new config
+    command_set = ('ap', CLEAR, setup_ipa, TimedCommand('y', 40), TimedCommand('n', 5))
+    configured_appliance.appliance_console.run_commands(command_set)
+    wait_for(lambda: not configured_appliance.sssd.running)
 
 
-@pytest.mark.uncollect('No IPA servers currently available')
-@pytest.mark.parametrize('auth_type', [
-    LoginOption('sso', 'sso_enabled', '1'),
-    LoginOption('saml', 'saml_enabled', '2'),
-    LoginOption('local_login', 'local_login_disabled', '3')
-], ids=['sso', 'saml', 'local_login'])
-def test_appliance_console_external_auth(auth_type, app_creds, ipa_crud):
-    """'ap' launches appliance_console, '' clears info screen, '12' change ext auth options,
+@pytest.mark.parametrize('auth_type', ext_auth_options, ids=[opt.name for opt in ext_auth_options])
+def test_appliance_console_external_auth(auth_type, app_creds, ipa_crud, configured_appliance):
+    """'ap' launches appliance_console, '' clears info screen, '12/15' change ext auth options,
     'auth_type' auth type to change, '4' apply changes."""
+    # TODO this depends on the auth_type options being disabled when the test is run
+    # TODO it assumes that first switch is to true, then false.
 
     evm_tail = LogValidator('/var/www/miq/vmdb/log/evm.log',
                             matched_patterns=['.*{} to true.*'.format(auth_type.option)],
-                            hostname=ipa_crud.hostname,
+                            hostname=configured_appliance.hostname,
                             username=app_creds['sshlogin'],
-                            password=app_creds['password'])
+                            password=app_creds['sshpass'])
     evm_tail.fix_before_start()
-    command_set = ('ap', '', '12', auth_type.index, '4')
-    ipa_crud.appliance_console.run_commands(command_set)
+    ext_auth = '12'
+    command_set = ('ap', '', ext_auth, auth_type.index, '4')
+    configured_appliance.appliance_console.run_commands(command_set)
```

Comment:
**QUESTION:** Shouldn't we assert on the return of the command being successful?

Reply:
Not possible in Appliance.ApplianceConsole.run_commands, unfortunately.